d2/src/test/java/com/linkedin/d2/balancer/servers/ZookeeperConnectionManagerTest.java

```diff
@@ -471,6 +472,311 @@ public void testMarkUpAndDownMultipleTimesFinalUp()
     executorService.shutdown();
   }
 
+  @Test
+  public void testNoWarmupWhenDisabled() throws Exception
```

Comment:
Thanks for writing all the unit tests. These are very comprehensive and convincing